docs/intro/overrides.rst

```diff
@@ -0,0 +1,387 @@
+        def to_item(self):
+            ...  # more specific parsing
+
+The code above declares that:
+
+    - For sites that matches the ``example.com`` pattern, ``ExampleProductPage``
+      would be used instead of ``GenericProductPage``.
+    - The same is true for ``YetAnotherExampleProductPage`` where it is used
+      instead of ``GenericProductPage`` for two URLs: ``dualexample.com`` and
+      ``dualexample.net``.
+    - However, ``AnotherExampleProductPage`` is only used instead of ``GenericProductPage``
+      when we're parsing pages from ``anotherexample.com`` which doesn't contain
+      ``/digital-goods/`` in its URL path.
+
+The override mechanism that ``web-poet`` offers could still be further
+customized. You can read some of the specific parameters and alternative ways
+to organize the rules via the :ref:`Overrides API section <api-overrides>`.
+
+To demonstrate another alternative way to declare the Override rules, see the
+code example below:
+
+.. code-block:: python
+
+    from web_poet.pages import ItemWebPage
+    from web_poet import PageObjectRegistry
+
+    primary_registry = PageObjectRegistry()
+    secondary_registry = PageObjectRegistry()
+
+    class GenericProductPage(ItemWebPage):
+        def to_item(self):
+            return {"product title": self.css("title::text").get()}
+
+    @primary_registry.handle_urls("example.com", overrides=GenericProductPage)
+    class ExampleProductPage(ItemWebPage):
+        def to_item(self):
+            ...  # more specific parsing
+
+    @secondary_registry.handle_urls("anotherexample.com", overrides=GenericProductPage, exclude="/digital-goods/")
+    class AnotherExampleProductPage(ItemWebPage):
+        def to_item(self):
+            ...  # more specific parsing
+
+    @primary_registry.handle_urls(["dualexample.com", "dualexample.net"], overrides=GenericProductPage)
+    @secondary_registry.handle_urls(["dualexample.com", "dualexample.net"], overrides=GenericProductPage)
+    class DualExampleProductPage(ItemWebPage):
+        def to_item(self):
+
+If you need more control over the Registry, you could instantiate your very
+own :class:`~.PageObjectRegistry` and use its ``@handle_urls`` to annotate and
+register the rules. This might benefit you in certain project use cases where you
+need more organizational control over your rules.
+
+Viewing all available Overrides
+-------------------------------
+
+A convenience function is available discover and retrieve all rules from your
+project. Make sure to check out :ref:`Overrides API section <api-overrides>`
+to see the other functionalities.
+
+.. code-block:: python
+
+    from web_poet import default_registry
+
+    # Retrieves all rules that were registered in the registry
+    rules = default_registry.get_overrides()
+
+    # Or, we could also filter out the rules by the module they were defined in
+    rules = default_registry.get_overrides_from_module("my_project.page_objects")
```

Comment:
This example doesn't have the same caveat as an example above, so it's actually quite different; it is not only about filtering, but also about making it actually work.

Reply:
Gotcha. 👍 

BTW, I updated the newly renamed `get_overrides_from()` _(formerly called `get_overrides_from_module()`)_ to accept an arbitrary number of strings for convenience. Updated this on https://github.com/scrapinghub/web-poet/pull/16/commits/3b05c075d1c49058139fb36ec7ec9982f3ab41a2

Reply:
Resolving this thread since the latest commits have combined the functionality of `get_overrides_from_module()` into one simple API named `get_overrides()` which has a **consume** and **filter** params.

More info in https://github.com/scrapinghub/web-poet/pull/16/files#diff-409cb5fe79e7d7d5ede41109573e31bd847a147a966f477de80c99279bcdf479R247-R251